src/curry/compare.ts

```diff
@@ -0,0 +1,61 @@
+import { CompatibleProperty, isFunction } from 'radashi'
+
+// Used to detect type `any`
+declare const AnyKind: unique symbol
+declare class Any {
+  private declare any: typeof AnyKind
+}
+
+export type Comparable =
+  | number
+  | string
```

Comment:
I'm not sure if `leftValue > rightValue ? 1 : leftValue < rightValue ? -1 : 0` is a good comparator for strings. We can use `localCompare` or not provide a default comparator for strings at all.

Reply:
The default comparator is intended to be both minimal and flexible (support comparing any two `Comparable` values). The caller can opt-in to `localeCompare` like [`alphabetical` is doing](https://github.com/radashi-org/radashi/pull/36/files#diff-54014b3c0f8b7db7e6ac41b359cca9652f2aebe6de0205bacb453ae1358a36e4R23).

I suppose we could include a warning about this behavior in the function's description.